cmake/modules/AddSwift.cmake

```diff
@@ -270,6 +276,13 @@ function(_add_variant_c_compile_flags)
         "-I${SWIFT_ANDROID_NDK_PATH}/sources/android/support/include")
   endif()
 
+  if("${CFLAGS_SDK}" STREQUAL "FUCHSIA")
+    list(APPEND result
+      "-mcx16"
+      "-I${SWIFT_FUCHSIA_BUILD_PATH}/third_party/icu/source/common/"
+      "-I${SWIFT_FUCHSIA_BUILD_PATH}/third_party/icu/source/i18n/")
+  endif()
```

Comment:
I don't recall all the details, but I believe Swift/Android started out with some hardcoded paths like this, and those were later refactored into something that most other platforms could use. Maybe it was #3205? I wonder if you could use that mechanism as well?

Reply:
Yeah, I was going to move to the SDK defined include paths for ICU in the top level cmake. There isn't a built-util arg to set those per SDK though. The android target added their own one off to override that though. I need to fix it because I think it's possible for your system's ICU headers can get accidentally included if nothing is set instead of the target platform's ICU headers when compiling the stdlib.

Flinging other args and paths up through build-util, the build-util-impl, and finally cmake and adding tests for it can be a bit of chore so I decided that since this path is well known on Fuchsia (unlike the android build where the user could checkout the ICU they use anywhere) that it would would ok temporarily to add it relative to the directory I had already flung up the layers. Having hardcoded paths like this though is generally bad.

Reply:
"-mcx16" is also interesting here. Having it (or any target CPU that implies that flag) might speed up refcounting heap objects on Linux and it will likely fix an issue FreeBSD users had with building swift unless they installed gcc on their system (or really anyone that is using compiler-rt and instead of libgcc/atomic). We noticed the issue since we use compiler-rt which doesn't have an implementation of "_sync_val_compare_and_swap_16" like libgcc/libatomic does. 

Since the swift cmake doesn't set any minimum target CPU, LLVM will assume it's ok in it's target lowering that you are CPU that doesn't support the CMPXCHG16B instruction and that it's ok to call out to that libgcc intrinsic. HeapObject.cpp is calling out to that function a lot on Linux which is probably a lot slower than the native instruction for that on x86 CPUs that support that instruction. 

See https://bugs.swift.org/browse/SR-5779

This might have also been partially fixed in LLVM recently.

Reply:
I'm reusing some of the settings from that change now (specifically just the include paths) so I can reuse some of the logic. The icu libraries in Fuchsia are in the same shared libs directory as number of our shared libraries that are not in our sysroot so I just use that path for search paths everywhere.